SMLHelper/Patchers/PingTypePatcher.cs

```diff
@@ -0,0 +1,151 @@
+
+        internal static PingType? GetPingType(string pingName)
+        {
+            try
+            {
+                var ping = Enum.Parse(typeof(PingType), pingName);
+                return (PingType) ping;
+            }
+            catch (ArgumentException e)
+            {
+                return new PingType?();
+            }
+        }
+        
+        internal static void Patch(HarmonyInstance harmony)
+        {
+            IngameMenuHandler.Main.RegisterOneTimeUseOnSaveEvent(() => cacheManager.SaveCache());
+
+            harmony.Patch(AccessTools.Method(typeof(Enum), nameof(Enum.GetValues)),
+                postfix: new HarmonyMethod(AccessTools.Method(typeof(PingTypePatcher), nameof(Postfix_GetValues))));
+
+            harmony.Patch(AccessTools.Method(typeof(Enum), nameof(Enum.IsDefined)),
+                prefix: new HarmonyMethod(AccessTools.Method(typeof(PingTypePatcher), nameof(Prefix_IsDefined))));
+
+            harmony.Patch(AccessTools.Method(typeof(Enum), nameof(Enum.Parse), new Type[] { typeof(Type), typeof(string), typeof(bool) }),
+                prefix: new HarmonyMethod(AccessTools.Method(typeof(PingTypePatcher), nameof(Prefix_Parse))));
+
+            harmony.Patch(AccessTools.Method(typeof(PingType), nameof(PingType.ToString), new Type[] { }),
+                prefix: new HarmonyMethod(AccessTools.Method(typeof(PingTypePatcher), nameof(Prefix_ToString))));
+            
+            harmony.Patch(AccessTools.Method(typeof(SpriteManager), nameof(SpriteManager.Get), new Type[] { typeof(SpriteManager.Group), typeof(string) }),
+                prefix: new HarmonyMethod(AccessTools.Method(typeof(PingTypePatcher), nameof(Prefix_SpriteManager_Get))));
+
+            Logger.Log($"Added {cacheManager.ModdedKeysCount} PingTypes succesfully into the game.");
+            Logger.Log("PingTypePatcher is done.", LogLevel.Debug);
+        }
+
+        private static bool Prefix_SpriteManager_Get(SpriteManager.Group group, string name, ref Atlas.Sprite __result)
+        {
+            if (group == SpriteManager.Group.Pings)
+            {
+                var pingType = GetPingType(name);
+                if (pingType.HasValue && cachedSprites.ContainsKey(pingType.Value))
+                {
+                    __result = cachedSprites[pingType.Value];
+                    return false;
+                }
+            }
+
+            return true;
+        }
+        
+        private static void Postfix_GetValues(Type enumType, ref Array __result)
+        {
+            if (enumType == typeof(PingType))
+            {
+                __result = __result.Cast<PingType>().Concat(cacheManager.ModdedKeys).ToArray();
+            }
+        }
+        
+        private static bool Prefix_IsDefined(Type enumType, object value, ref bool __result)
+        {
+            var result = enumType == typeof(PingType) && cacheManager.ContainsKey((PingType) value);
+            __result |= result;
+            return !result;
```

Comment:
I can see what you're doing here, but it's less readable and harder to follow when "result" is used in multiple ways like this.  
Can we go with a simpler style?
```CS
if (enumType == typeof(PingType) && cacheManager.ContainsKey((PingType) value))
{
    __result = true;
    return false;
}

return true;
```